### PR TITLE
[OPSRC-640]Plugin development

### PR DESCRIPTION
**Pull request by liszkapawel**
This PR repairs problems reported in comments at #11

### CODE REVIEW

File: tests/Behat/Context/Setup/GroupsContext.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
```
Comment (senghe) [inline tests/Behat/Context/Setup/GroupsContext.php:3]: No OS header found :)

File: tests/Behat/Context/Ui/GroupsContext.php [hunk 2]
```diff
@@ -74,55 +57,31 @@ public function __construct(
...
-    public function iSetItsNameTo(string $group): void
+    public function iSelectAttributeGroup(): void
     {
-        $this->createPage->fillName($group);
+        $this->updateSimpleProductPage->selectAttributeGroup();
     }
 
     /**
-     * @When I add it
+     * @Then I should see button to add AttributesGroup
```
Comment (senghe) [inline tests/Behat/Context/Ui/GroupsContext.php:68]:
```suggestion
     * @Then I should see button to add attribute group
```

File: tests/Behat/Context/Ui/GroupsContext.php [hunk 3]
```diff
@@ -74,55 +57,31 @@ public function __construct(
...
+    public function iShouldSeeButtonToAddAttributesGroup(): void
     {
-        $this->createPage->create();
+        $buttonAddAttributesGroupName = $this->updateSimpleProductPage->getButtonToAddAttributesGroup();
+        assertEquals('add_attributes_group', $buttonAddAttributesGroupName);
     }
 
     /**
-     * @When I assign this attributes to group:
+     * @Given these attributes should be visible next to the group :group:
```
Comment (senghe) [inline tests/Behat/Context/Ui/GroupsContext.php:77]:
```suggestion
     * @Given these attributes should be visible next to the :group: group
```

File: tests/Behat/Context/Ui/GroupsContext.php [hunk 4]
```diff
@@ -142,89 +101,48 @@ public function theGroupShouldAppearInTheStore(string $group): void
...
     {
-        $this->updateSimpleProductPage->openTab($tabName);
+        $codes = array_merge([], ...$table->getRows());
+
+        $this->createGroupPage->assignAttributes($codes);
     }
 
     /**
-     * @When I select attribute group
+     * @Given there is created group with name :group
```
Comment (senghe) [inline tests/Behat/Context/Ui/GroupsContext.php:141]:
```suggestion
     * @Given there is created group with :group name
```

File: tests/Behat/Context/Ui/GroupsContext.php [hunk 5]
```diff
@@ -248,31 +166,35 @@ public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
...
+        $product = $this->productRepository->findOneByCode($productName);
+
+        $this->sharedStorage->set('product', $product);
+        if ($product->isSimple()) {
+            $this->testHelper->waitUntilPageOpens($this->baseUpdateSimpleProductPage, ['id' => $product->getId()]);
+
+            return;
+        }
+
+        $this->testHelper->waitUntilPageOpens($this->baseUpdateSimpleProductPage, ['id' => $product->getId()]);
```
Comment (senghe) [inline tests/Behat/Context/Ui/GroupsContext.php:190]: What's the difference between the two page openings?

File: tests/Behat/Page/Admin/Product/UpdateSimpleProductPageInterface.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (senghe) [inline tests/Behat/Page/Admin/Product/UpdateSimpleProductPageInterface.php:1]: No OS header :)

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Behat/Context/Ui/GroupsContext.php (hunk 5 above)
Comment (liszkapawel) [reply tests/Behat/Context/Ui/GroupsContext.php:190]: changed

File: tests/Behat/Context/Ui/GroupsContext.php (hunk 3 above)
Comment (liszkapawel) [reply tests/Behat/Context/Ui/GroupsContext.php:77]: changed

File: tests/Behat/Context/Ui/GroupsContext.php (hunk 4 above)
Comment (liszkapawel) [reply tests/Behat/Context/Ui/GroupsContext.php:141]: changed

File: tests/Behat/Context/Setup/GroupsContext.php (hunk 1 above)
Comment (liszkapawel) [reply tests/Behat/Context/Setup/GroupsContext.php:3]: changed

File: tests/Behat/Page/Admin/Product/UpdateSimpleProductPageInterface.php (hunk 6 above)
Comment (liszkapawel) [reply tests/Behat/Page/Admin/Product/UpdateSimpleProductPageInterface.php:1]: changed

File: tests/Behat/Context/Ui/GroupsContext.php (hunk 2 above)
Comment (liszkapawel) [reply tests/Behat/Context/Ui/GroupsContext.php:74]: changed

**Comment by christopherhero** [comment]
**FRONTEND** - approved (https://github.com/BitBagCommerce/SyliusProductAttributeGroupsPlugin/pull/14)

**Pull request merged**